crates/notion-core/src/shim.rs

```diff
@@ -2,29 +2,18 @@
 
 use std::{fs, io};
 
-use failure::Fail;
-
+use crate::error::ErrorDetails;
 use crate::path;
-use notion_fail::{throw, ExitCode, FailExt, Fallible, NotionFail};
-use notion_fail_derive::*;
-
-#[derive(Debug, Fail, NotionFail)]
-#[fail(display = "{}", error)]
-#[notion_fail(code = "FileSystemError")]
-pub(crate) struct SymlinkError {
-    error: String,
-}
+use notion_fail::{throw, FailExt, Fallible};
 
-impl SymlinkError {
-    pub(crate) fn from_io_error(error: &io::Error) -> Self {
-        if let Some(inner_err) = error.get_ref() {
-            SymlinkError {
-                error: inner_err.to_string(),
-            }
-        } else {
-            SymlinkError {
-                error: error.to_string(),
-            }
+fn from_io_error(error: &io::Error) -> ErrorDetails {
```

Comment:
Use a name like `symlink_error` instead, since you're trying to express what kind of context you're providing to `with_context`.